extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/AbstractMethodsAdder.java

```diff
@@ -203,10 +203,21 @@ protected void generateFindQueryResultHandling(MethodCreator methodCreator, Resu
 
             methodCreator.returnValue(sliceResult);
 
+        } else if (isIntLongOrBoolean(returnType)) {
+            ResultHandle singleResult = methodCreator.invokeInterfaceMethod(
+                    MethodDescriptor.ofMethod(PanacheQuery.class, "singleResult", Object.class),
+                    panacheQuery);
+            methodCreator.returnValue(singleResult);
         } else {
             throw new IllegalArgumentException(
                     "Return type of method " + methodName + " of Repository " + repositoryClassInfo
                             + " does not match find query type");
         }
     }
+
+    protected boolean isIntLongOrBoolean(DotName dotName) {
+        return DotNames.BOOLEAN.equals(dotName) || DotNames.PRIMITIVE_BOOLEAN.equals(dotName)
+                || DotNames.INTEGER.equals(dotName) || DotNames.PRIMITIVE_INTEGER.equals(dotName)
+                || DotNames.LONG.equals(dotName) || DotNames.PRIMITIVE_LONG.equals(dotName);
```

Comment:
Shouldn't we consider all the primitive types here while we are at it?

Reply:
We potentially could, but I don't know if they are all support by Spring Data anyway. These are the ones that show up often.